docs/specs/code/SIGNING.md

```diff
@@ -0,0 +1,183 @@
+
+# Motivation
+
+Most common use cases:
+
+1) Proving ownership of an address/funds
+1) Proving ownership of addresses used in a transaction
+1) Proving owernship of an identity or other off-chain data with a public key attached to it
+
+# Background
+
+We make use of the following existing work
+
+### Cryptography
+
+Cardano already allows message signing within the WASM bindings. Notably,
+
+1) [sign](https://github.com/input-output-hk/js-cardano-wasm/blob/master/cardano-wallet/src/lib.rs#L177)
+2) [verify](https://github.com/input-output-hk/js-cardano-wasm/blob/master/cardano-wallet/src/lib.rs#L232)
+
+You can see an example of these two functions [here](https://repl.it/repls/GlossyDarkgreenMethod)
+
+### Known complexities
+
+Even if you use cryptographically sound `sign` and `verify` functions, you still have the following problems:
+
+1) No human-recognizable prefix
+0) No error detection
+0) User could accidentally sign a transaction or a block thinking it's harmless data
+
+We also have a risk of a few different kinds of replay attacks
+
+4) A dapp asks person A to sign "BOB" and then another dapp asks user B to sign "BOB". B can just use the signature from A
+0) A dapp asks person A to sign "BOB" on testnet chain. Person B then sends this signed message to the same dapp running on mainnet (same argument applies to sidechains)
+
+Additionally, signing for hardware wallets has additional difficulties such as
+
+6) User needs to be able to verify input & output data on device
+0) Device has limited memory to store input data
+
+### Other standards
+
+Other blockchains have existing specifications for message signing, but they mostly revolve around scripts trying to validate messages. We don't leverage any of their work in particular but it may be of interest.
+
+- [BIP-137](https://github.com/bitcoin/bips/blob/master/bip-0137.mediawiki) - simply scheme for message signing that works with P2PKH, P2PSH and bech32
+- [BIP-322](https://github.com/kallewoof/bips/blob/master/bip-0322.mediawiki) - reuses Bitcoin script to process a generic signed message format
+
+- [EIP-191](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-191.md) - encode data for Ethereum contracts
+- [EIP-712](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md) - encode structs for Ethereum contracts
+
+# Proposal
+
+We propose a scheme that is generic in nature. By that we mean it does not specify a serialization format for the actual data you wish to sign nor does it enforce a specific communication channel between the signer and the validator.
+
+For the encoding, we will use [bech32](https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki) not only because it solves problems (1), (2) and (3) but because it will already be implemented for Cardano starting in Shelley as specified in [CIP-001](https://github.com/input-output-hk/implementation-decisions/blob/master/text/0001-address.md
```

Comment:
maybe specify that "encoding" here refers to signature encoding, not data encoding